subprojects/core/src/main/java/org/gradle/execution/plan/FinalizerGroup.java

```diff
@@ -22,6 +22,7 @@
 import com.google.common.collect.SetMultimap;
 
 import javax.annotation.Nullable;
+import java.util.ArrayDeque;
```

Comment:
This was using `kotlin.collection.ArrayDeque`
Was it intentional?

That type is not available in 7.x so I replaced it with the `java.util` version here.